src/test/script_tests.cpp

```diff
@@ -943,12 +943,15 @@ BOOST_AUTO_TEST_CASE(script_json_test)
             unsigned int i=0;
             for (i = 0; i < test[pos].size()-1; i++) {
                 auto element = test[pos][i].get_str();
-                // Taproot script - third from the last element in witness stack
-                if (test[pos].size() >= 3 && i == test[pos].size()-3 && !IsHex(element)) {
-                    witness.stack.push_back(ToByteVector(ParseScript(element)));
-                // Taproot script control block - second from the last element in witness stack
-                // If <AUTOGEN:CONTROLBLOCK> we auto-generate the control block
-                } else if (test[pos].size() >= 3 && i == test[pos].size()-2 && strcmp(element.c_str(), "<AUTOGEN:CONTROLBLOCK>") == 0) {
+                // We use #SCRIPT# to flag a non-hex script
+                std::string scriptFlag = std::string("#SCRIPT#");
+                if (element.find(scriptFlag) == 0) {
```

Comment:
is interesting the `find` function returns 0 if there is a find. Or maybe it returns the index of the match

Reply:
You are correct!

> The position of the first character of the first match.
 If no matches were found, the function returns [string::npos](https://cplusplus.com/string::npos).

https://cplusplus.com/reference/string/string/find/